hazelcast/src/main/java/com/hazelcast/map/impl/wan/WANContext.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.wan;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.WanConsumerConfig;
+import com.hazelcast.config.WanReplicationConfig;
+import com.hazelcast.config.WanReplicationRef;
+import com.hazelcast.config.WanSyncConfig;
+import com.hazelcast.jet.impl.util.ConcurrentMemoizingSupplier;
+import com.hazelcast.map.impl.MapContainer;
+import com.hazelcast.map.impl.MapServiceContext;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.spi.merge.SplitBrainMergePolicy;
+import com.hazelcast.spi.merge.SplitBrainMergePolicyProvider;
+import com.hazelcast.wan.impl.DelegatingWanScheme;
+import com.hazelcast.wan.impl.WanReplicationService;
+
+import static com.hazelcast.config.ConsistencyCheckStrategy.MERKLE_TREES;
+import static com.hazelcast.internal.config.MergePolicyValidator.checkMapMergePolicy;
+import static java.lang.Boolean.TRUE;
+
+public class WANContext {
+
+    protected volatile SplitBrainMergePolicy wanMergePolicy;
+    protected ConcurrentMemoizingSupplier<DelegatingWanScheme> wanReplicationDelegateSupplier;
+    private final MapConfig mapConfig;
+    private final String name;
+    private final MapServiceContext mapServiceContext;
+    private volatile boolean persistWanReplicatedData;
+
+    public WANContext(MapContainer mapContainer) {
+        this.mapConfig = mapContainer.getMapConfig();
+        this.name = mapContainer.getName();
+        this.mapServiceContext = mapContainer.getMapServiceContext();
+    }
+
+    public void start() {
+        initWanReplication();
+    }
+
+    private void initWanReplication() {
+        NodeEngine nodeEngine = mapServiceContext.getNodeEngine();
+        WanReplicationRef wanReplicationRef = mapConfig.getWanReplicationRef();
+        if (wanReplicationRef == null) {
+            return;
+        }
+        String wanReplicationRefName = wanReplicationRef.getName();
+
+        Config config = nodeEngine.getConfig();
+        if (!TRUE.equals(mapConfig.getMerkleTreeConfig().getEnabled())
```

Comment:
I'm curious about the reasoning for `!TRUE.equals(mapConfig.getMerkleTreeConfig().getEnabled())` instead of just `!mapConfig.getMerkleTreeConfig().getEnabled()`?